### ISSUE TITLE
Bug: `Array unpacking can be simplified` incorrecly called on non-arrays

### ISSUE THREAD

**Issue by asgraf** [hammer-tools/php-hammer#7]
```php
function someFunctionReturningIterator(): \Generator
{
  yield 'a';
  yield 'b';
  yield 'c';
}

$variable = [...someFunctionReturningIterator()];//this line casts iterable to array
```
Code above currently incorrectly triggers [Array unpacking can be simplified](https://github.com/hammer-tools/php-hammer/wiki/Inspections#array-unpacking-can-be-simplified)
In this example `$variable` has type of `array`.
Applying [Array unpacking can be simplified](https://github.com/hammer-tools/php-hammer/wiki/Inspections#array-unpacking-can-be-simplified) fix would change  `$variable` type from `array` back to `\Generator`

**Comment by rentalhost** [comment]
Thanks for the report. The problem makes perfect sense and in fact, in this specific case, simplification cannot be suggested for `\Generators`. I'm going to do the repair today.

**Comment by rentalhost** [comment]
**Implemented to version 0.21.0:** it takes about three days to be published on the market. But if you want, you can download it from **[Releases](https://github.com/hammer-tools/php-hammer/releases)**.
